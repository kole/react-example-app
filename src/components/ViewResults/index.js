import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { inject, observer } from 'mobx-react';

import YourScore from '~/components/YourScore';
import Button from '~/components/Button';
import ResultsList from '~/components/ResultsList';

const ResultsView = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    min-height: 100%;
    padding: 4em;
    text-align: center;

    > * {
        flex: 1;
        flex-direction: column;
        justify-content: center;
    }
`;

@inject('rootStore')
@observer
export default class Results extends React.Component {
    componentDidMount() {
        const { rootStore: { QuestionsStore: { questions }, AnswersStore: { answers }}, history} = this.props;
        if (questions.length === 0 || questions.length !== answers.length) {
            return history.push('/');
        }
    }

    playAgain() {
        const { history } = this.props;
        return history.push('/');
    }

    render() {
        const {
            rootStore: {
                QuestionsStore: { questions },
                AnswersStore: { answers },
                ResultsStore: { correctAnswerCount }
            }
        } = this.props;

        return (
            <ResultsView>
                <YourScore correctAnswerCount={correctAnswerCount} totalQuestions={questions.length} />
                <ResultsList answers={answers} questions={questions} />
                <Button clickHandler={this.playAgain.bind(this)} text="Play Again?" />
            </ResultsView>
        );
    }
}

Results.propTypes = {
    rootStore: PropTypes.object,
    history: PropTypes.object.isRequired
}
