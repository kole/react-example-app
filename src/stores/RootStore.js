import Questions from './questions';
import Answers from './answers';
import Results from './results';

export default class RootStoreClass {
    constructor() {
        this.QuestionsStore = new Questions(this);
        this.AnswersStore = new Answers(this);
        this.ResultsStore = new Results(this);
    }
}
