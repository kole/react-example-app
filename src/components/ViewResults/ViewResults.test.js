/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import ViewResults from './index.js';
import RootStore from '~/stores/RootStore';
const store = new RootStore();

const mocks = {
    history: {push: () => {}}
};

describe('ViewResults', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<ViewResults history={mocks.history} rootStore={store} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
