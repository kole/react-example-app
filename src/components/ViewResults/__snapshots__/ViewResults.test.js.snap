// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewResults Snapshot 1`] = `
<div
  className="sc-EHOje gQKliD"
>
  <div
    className="sc-bdVaJa krsHcM"
  >
    You scored 
    <br />
    0
     / 
    0
  </div>
  <div
    className="sc-ifAKCX llzVb"
  >
    <ul />
  </div>
  <div
    className="sc-bwzfXH dMNwrz"
  >
    <button
      onClick={[Function]}
    >
      Play Again?
    </button>
  </div>
</div>
`;
